diff --git a/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/helpers/dataTypeCodeHelper.js b/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/helpers/dataTypeCodeHelper.js
index 5ab1158..23a1174 100644
--- a/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/helpers/dataTypeCodeHelper.js
+++ b/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/helpers/dataTypeCodeHelper.js
@@ -16,6 +16,8 @@ export function getArrayConstructor(datatypeCode) {
             return Uint16Array;
         case NIFTICONSTANTS.NIFTI_TYPE_UINT32:
             return Uint32Array;
+        case NIFTICONSTANTS.NIFTI_TYPE_FLOAT64:
+            return Float64Array;
         default:
             throw new Error(`NIFTI datatypeCode ${datatypeCode} is not yet supported`);
     }
diff --git a/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/helpers/modalityScaleNifti.js b/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/helpers/modalityScaleNifti.js
index e91e6c6..a91c0a6 100644
--- a/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/helpers/modalityScaleNifti.js
+++ b/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/helpers/modalityScaleNifti.js
@@ -70,6 +70,10 @@ export default function modalityScaleNifti(niftiHeader, niftiImageBuffer) {
             niiBuffer = new Uint32Array(niftiImageBuffer);
             scalarData = allocateScalarData('Float32Array', niiBuffer);
             break;
+        case NIFTICONSTANTS.NIFTI_TYPE_FLOAT64:
+            niiBuffer = new Float64Array(niftiImageBuffer);
+            scalarData = allocateScalarData('Float64Array', niiBuffer);
+            break;
         default:
             throw new Error(`NIFTI datatypeCode ${datatypeCode} is not yet supported`);
     }
@@ -127,6 +131,11 @@ function allocateScalarData(types, niiBuffer) {
             checkCacheAvailable(bitsAllocated, nVox);
             scalarData = new Uint8Array(nVox);
             break;
+        case 'Float64Array':
+            bitsAllocated = 64;
+            checkCacheAvailable(bitsAllocated, nVox);
+            scalarData = new Float64Array(nVox);
+            break;
         default:
             throw new Error(`TypedArray ${types} is not yet supported`);
     }
diff --git a/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/cornerstoneNiftiImageLoader.js b/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/cornerstoneNiftiImageLoader.js
index a56de79..4450c83 100644
--- a/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/cornerstoneNiftiImageLoader.js
+++ b/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/cornerstoneNiftiImageLoader.js
@@ -4,22 +4,22 @@ import { Events } from './enums';
 import { modalityScaleNifti } from './helpers';
 import { getOptions } from './internal';
 const dataFetchStateMap = new Map();
-function fetchArrayBuffer({ url, signal, onload, }) {
+async function fetchArrayBuffer({ url, signal, onload, }) {
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', url, true);
+    const defaultHeaders = {};
+    const options = getOptions();
+    const beforeSendHeaders = await options.beforeSend(xhr, defaultHeaders, url);
+    const headers = Object.assign({}, defaultHeaders, beforeSendHeaders);
+    xhr.responseType = 'arraybuffer';
+    Object.keys(headers).forEach(function (key) {
+        if (headers[key] === null) {
+            return;
+        }
+        xhr.setRequestHeader(key, headers[key]);
+    });
     return new Promise((resolve, reject) => {
-        const xhr = new XMLHttpRequest();
-        xhr.open('GET', url, true);
-        const defaultHeaders = {};
-        const options = getOptions();
-        const beforeSendHeaders = options.beforeSend(xhr, defaultHeaders, url);
-        const headers = Object.assign({}, defaultHeaders, beforeSendHeaders);
-        xhr.responseType = 'arraybuffer';
-        Object.keys(headers).forEach(function (key) {
-            if (headers[key] === null) {
-                return;
-            }
-            xhr.setRequestHeader(key, headers[key]);
-        });
-        const onLoadHandler = function (e) {
+        const onLoadHandler = function (_e) {
             if (onload && typeof onload === 'function') {
                 onload();
             }
@@ -111,9 +111,32 @@ function waitForNiftiData(imageId, url, sliceIndex, imagePixelModule, imagePlane
 function createImage(imageId, sliceIndex, imagePixelModule, imagePlaneModule, niftiScalarData) {
     const { rows, columns } = imagePlaneModule;
     const numVoxels = rows * columns;
-    const sliceOffset = numVoxels * sliceIndex;
     const pixelData = new niftiScalarData.constructor(numVoxels);
-    pixelData.set(niftiScalarData.subarray(sliceOffset, sliceOffset + numVoxels));
+    const niftiHeaderMeta = metaData.get('niftiHeader', imageId);
+    const sliceDimIndex = niftiHeaderMeta?.sliceDimIndex ?? 2;
+    const niftiHdr = niftiHeaderMeta?.header;
+    if (sliceDimIndex === 2 || !niftiHdr) {
+        const sliceOffset = numVoxels * sliceIndex;
+        pixelData.set(niftiScalarData.subarray(sliceOffset, sliceOffset + numVoxels));
+    }
+    else if (sliceDimIndex === 1) {
+        const dimI = niftiHdr.dims[1];
+        const dimJ = niftiHdr.dims[2];
+        for (let k = 0; k < rows; k++) {
+            const srcOffset = sliceIndex * dimI + k * dimI * dimJ;
+            pixelData.set(niftiScalarData.subarray(srcOffset, srcOffset + dimI), k * dimI);
+        }
+    }
+    else {
+        const dimI = niftiHdr.dims[1];
+        const dimJ = niftiHdr.dims[2];
+        for (let k = 0; k < rows; k++) {
+            for (let j = 0; j < columns; j++) {
+                pixelData[k * columns + j] =
+                    niftiScalarData[sliceIndex + j * dimI + k * dimI * dimJ];
+            }
+        }
+    }
     const voxelManager = utilities.VoxelManager.createImageVoxelManager({
         width: columns,
         height: rows,
diff --git a/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/createNiftiImageIdsAndCacheMetadata.js b/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/createNiftiImageIdsAndCacheMetadata.js
index cb78a63..d7c5ba1 100644
--- a/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/createNiftiImageIdsAndCacheMetadata.js
+++ b/node_modules/@cornerstonejs/nifti-volume-loader/dist/esm/createNiftiImageIdsAndCacheMetadata.js
@@ -2,7 +2,6 @@ import * as NiftiReader from 'nifti-reader-js';
 import { eventTarget, triggerEvent, utilities } from '@cornerstonejs/core';
 import { rasToLps } from './helpers/convert';
 import Events from './enums/Events';
-import { NIFTI_LOADER_SCHEME } from './constants';
 import makeVolumeMetadata from './helpers/makeVolumeMetadata';
 import { getArrayConstructor } from './helpers/dataTypeCodeHelper';
 import { getOptions } from './internal';
@@ -123,6 +122,13 @@ function handleNiftiHeader(data) {
         const version = header.sizeof_hdr === NIFTI2_HEADER_SIZE ? 2 : 1;
         const { orientation, origin, spacing } = rasToLps(header);
         const { dimensions, direction } = makeVolumeMetadata(header, orientation, 1);
+        let sliceDimIndex = 2;
+        if (spacing[1] > spacing[0] && spacing[1] > spacing[2]) {
+            sliceDimIndex = 1;
+        }
+        else if (spacing[0] > spacing[1] && spacing[0] > spacing[2]) {
+            sliceDimIndex = 0;
+        }
         const arrayConstructor = getArrayConstructor(header.datatypeCode);
         return {
             dimensions,
@@ -135,6 +141,7 @@ function handleNiftiHeader(data) {
             spacing,
             header,
             arrayConstructor,
+            sliceDimIndex,
         };
     }
     catch (error) {
@@ -163,50 +170,69 @@ async function fetchAndAllocateNiftiVolume(url) {
             onHeader: resolve,
         });
     }));
-    const { dimensions, direction, isValid, message, origin, version, header, spacing, arrayConstructor, } = niftiHeader;
-    const numImages = dimensions[2];
+    const { dimensions, direction, isValid, message, origin, version, header, spacing, arrayConstructor, sliceDimIndex, } = niftiHeader;
     if (!isValid) {
         console.error(message);
         return;
     }
+    let rowAxisIdx, colAxisIdx;
+    if (sliceDimIndex === 0) {
+        rowAxisIdx = 1;
+        colAxisIdx = 2;
+    }
+    else if (sliceDimIndex === 1) {
+        rowAxisIdx = 0;
+        colAxisIdx = 2;
+    }
+    else {
+        rowAxisIdx = 0;
+        colAxisIdx = 1;
+    }
+    const dirVec = [
+        Array.from(direction.slice(0, 3)),
+        Array.from(direction.slice(3, 6)),
+        Array.from(direction.slice(6, 9)),
+    ];
+    const numImages = dimensions[sliceDimIndex];
+    const imageCols = dimensions[rowAxisIdx];
+    const imageRows = dimensions[colAxisIdx];
+    const sliceSpacing = spacing[sliceDimIndex];
+    const colSpacing = spacing[rowAxisIdx];
+    const rowSpacing = spacing[colAxisIdx];
+    const rowCosinesVec = dirVec[rowAxisIdx];
+    const colCosinesVec = dirVec[colAxisIdx];
+    const scanAxisVec = dirVec[sliceDimIndex];
     const imageIds = [];
     for (let i = 0; i < numImages; i++) {
         const imageId = `nifti:${niftiURL}?frame=${i}`;
         const imageIdIndex = i;
         imageIds.push(imageId);
-        const imageOrientationPatient = [
-            direction[0],
-            direction[1],
-            direction[2],
-            direction[3],
-            direction[4],
-            direction[5],
-        ];
+        const imageOrientationPatient = [...rowCosinesVec, ...colCosinesVec];
         const precision = 6;
         const imagePositionPatient = [
-            parseFloat((origin[0] + imageIdIndex * direction[6] * spacing[0]).toFixed(precision)),
-            parseFloat((origin[1] + imageIdIndex * direction[7] * spacing[1]).toFixed(precision)),
-            parseFloat((origin[2] + imageIdIndex * direction[8] * spacing[2]).toFixed(precision)),
+            parseFloat((origin[0] + imageIdIndex * scanAxisVec[0] * sliceSpacing).toFixed(precision)),
+            parseFloat((origin[1] + imageIdIndex * scanAxisVec[1] * sliceSpacing).toFixed(precision)),
+            parseFloat((origin[2] + imageIdIndex * scanAxisVec[2] * sliceSpacing).toFixed(precision)),
         ];
         const imagePlaneMetadata = {
             frameOfReferenceUID: '1.2.840.10008.1.4',
-            rows: dimensions[1],
-            columns: dimensions[0],
+            rows: imageRows,
+            columns: imageCols,
             imageOrientationPatient,
-            rowCosines: direction.slice(0, 3),
-            columnCosines: direction.slice(3, 6),
+            rowCosines: rowCosinesVec,
+            columnCosines: colCosinesVec,
             imagePositionPatient,
-            sliceThickness: spacing[2],
-            sliceLocation: origin[2] + i * spacing[2],
-            pixelSpacing: [spacing[0], spacing[1]],
-            rowPixelSpacing: spacing[1],
-            columnPixelSpacing: spacing[0],
+            sliceThickness: sliceSpacing,
+            sliceLocation: imageIdIndex * sliceSpacing,
+            pixelSpacing: [rowSpacing, colSpacing],
+            rowPixelSpacing: rowSpacing,
+            columnPixelSpacing: colSpacing,
         };
         const imagePixelMetadata = {
             samplesPerPixel: 1,
             photometricInterpretation: 'MONOCHROME2',
-            rows: dimensions[1],
-            columns: dimensions[0],
+            rows: imageRows,
+            columns: imageCols,
             bitsAllocated: arrayConstructor.BYTES_PER_ELEMENT * 8,
             bitsStored: arrayConstructor.BYTES_PER_ELEMENT * 8,
             highBit: arrayConstructor.BYTES_PER_ELEMENT * 8 - 1,
@@ -248,6 +274,7 @@ async function fetchAndAllocateNiftiVolume(url) {
             type: 'niftiHeader',
             metadata: {
                 header,
+                sliceDimIndex,
             },
         });
     }
